src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/InterchangeHeader.java

```diff
@@ -0,0 +1,58 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A specialisation of a segment for the specific use case of an interchange header
+ * takes in specific values required to generate an interchange header
+ * example: UNB+UNOA:2+TES5+XX11+920113:1317+00000002'
+ */
+@Getter @Setter @RequiredArgsConstructor
+public class InterchangeHeader extends Segment {
+
+    private static DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyMMdd:hhmm");
+
+    private @NonNull String sender;
+    private @NonNull String recipient;
+    private @NonNull ZonedDateTime translationTime;
+    private Integer sequenceNumber;
+
+    @Override
+    public String getKey() {
+        return "UNB";
+    }
+
+    @Override
+    public String getValue() {
+        String timestamp = translationTime.format(DATE_FORMAT);
+        String formattedSequenceNumber = String.format("%08d", sequenceNumber);
+        return "UNOA:2"+"+"+sender+"+"+recipient+"+"+timestamp+"+"+formattedSequenceNumber;
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
+        if (sequenceNumber == null) {
```

Comment:
might be one if:
        if (Objects.nonNull(sequenceNumber) && sequenceNumber <= 0) {
            throw new EdifactValidationException(getKey() + ": Attribute sequenceNumber is required");
        }